twilight-gateway/src/shard.rs

```diff
@@ -867,13 +867,19 @@ impl Shard {
             Some(OpCode::Hello) => {
                 let event = Self::parse_event::<Hello>(buffer)?;
                 let interval = event.data.heartbeat_interval;
-                let heartbeat_duration = Duration::from_millis(interval);
-                self.heartbeat_interval = Some(heartbeat_duration);
 
                 if self.config().ratelimit_messages() {
                     self.ratelimiter = Some(CommandRatelimiter::new(interval));
                 }
 
+                let period = Duration::from_millis(interval);
+                // First heartbeat should have some jitter, see
+                // https://discord.com/developers/docs/topics/gateway#heartbeat-interval
+                let start = Instant::now() + period.mul_f64(rand::random());
+                let mut interval = time::interval_at(start, period);
+                interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
+                self.heartbeat_interval = Some(interval);
```

Comment:
imo a large grouping of lines can be hard to read 
```suggestion
                let period = Duration::from_millis(interval);

                // First heartbeat should have some jitter, see
                // https://discord.com/developers/docs/topics/gateway#heartbeat-interval
                let start = Instant::now() + period.mul_f64(rand::random());

                let mut interval = time::interval_at(start, period);
                interval.set_missed_tick_behavior(MissedTickBehavior::Delay);

                self.heartbeat_interval = Some(interval);
```